dialer.go

```diff
@@ -115,15 +116,19 @@ type connectionInfoCache interface {
 type monitoredCache struct {
 	openConns *uint64
 
+	mu          sync.Mutex
+	openSockets []*instrumentedConn
```

Comment:
The underlying Go interface is net.Conn. Let's use the same abstraction and call this `openConns`.